src/spacy/fortex/spacy/spacy_processors.py

```diff
@@ -575,23 +612,34 @@ def process_umls_entity_linking(linker, result, input_pack: DataPack):
 
     # get medical entity mentions and UMLS concepts
     for item in medical_entities:
-        entity = MedicalEntityMention(
-            input_pack, item.start_char, item.end_char
+        medical_entry_name = config.medical_onto_type
+        medical_entry = get_class(medical_entry_name)
+        entity = medical_entry(
+            pack=input_pack,
+            begin=item.start_char,
+            end=item.end_char,
         )
-        entity.ner_type = item.label_
+
+        setattr(entity, "ner_type", item.label_)
+        umls_entries = []
 
         for umls_ent in item._.kb_ents:
-            cui = umls_ent[0]
-            score = str(umls_ent[1])
+            cui_entity = linker.kb.cui_to_entity[umls_ent[0]]
+            umls = {}
+            umls["cui"] = umls_ent[0]
+            umls["score"] = str(umls_ent[1])
+            umls["name"] = cui_entity.canonical_name
+            umls["definition"] = cui_entity.definition
+            umls["tuis"] = cui_entity.types
+            umls["aliases"] = cui_entity.aliases
+
+            umls_entry_name = get_class(config.umls_onto_type)
+            umls_entry = umls_entry_name(pack=input_pack)
 
-            cui_entity = linker.kb.cui_to_entity[cui]
+            for attribute, _ in vars(umls_entry).items():
+                if attribute in umls.keys():
+                    setattr(umls_entry, attribute, umls[attribute])
 
-            umls = UMLSConceptLink(input_pack)
-            umls.cui = cui
-            umls.score = score
-            umls.name = cui_entity.canonical_name
-            umls.definition = cui_entity.definition
-            umls.tuis = cui_entity.types
-            umls.aliases = cui_entity.aliases
+            umls_entries.append(umls_entry)
 
-            entity.umls_entities.append(umls)
+        setattr(entity, "umls_entities", umls_entries)
```

Comment:
I think we cannot do `setattr` here since the list in forte entries are not normal python lists (https://github.com/asyml/forte/blob/master/ftx/onto/clinical.py#L69). This is because we actually need to store reference instead of the actual content. So we should probably do something like `getattr(entity, "umls_entities").append(umls_entry))` (Need to try it out to see if it works).

Reply:
i also realize the test case didn't test this part: https://github.com/asyml/forte-wrappers/blob/main/tests/wrappers/spacy_processors_test.py, maybe we should add a small test here?

Reply:
I have tested the current code, and it does work as expected. I see the `umls_entities` list populated properly. Maybe because `Flist` and native lists, both are mutable sequences, we are able to append to `Flist` through setattr too. I can try using and testing `getattr(entity, "umls_entities").append(umls_entry))` as well. 

I will add the missing test case for this use case.

Reply:
Added "umls_link" working to the existing test cases. Other tests run within seconds, but this one goes on for a very long time on my local machine. And I don't think tests are enabled in forte-wrapper's CI pipeline?